scripts/irods/test/test_ichmod.py

```diff
@@ -207,6 +207,43 @@ class Access(Enum):
                         if access_level.value in range(check[0].value, check[1].value + 1):
                             self.user.assert_icommand(params.get('cmd'), *check[2])
 
+    def test_ichmod_does_not_bypass_permission_model_for_data_object__issue_6579(self):
+        data_object = 'testFile'
+        user_permissions = ['null', 'read', 'write']
+
+        for permission in user_permissions:
+            with self.subTest(permission):
+                user_cur_dir, _, _ = self.user.run_icommand('ipwd')
+                self.user.assert_icommand(f'itouch {data_object}')
```

Comment:
not sure on our current 'best practices' on f-strings here or still with list notation (see slightly up or down in this file) to protect against bashisms (whitespace, etc.)